### PR TITLE
feat(aws): add `--profile` option support

### LINKED ISSUE #542: [aws] - Suggest AWS profiles
### Sanity checks

- [X] My issue relates to a specific CLI completion spec (e.g. `git checkout` is missing options in `git` completion spec). If your issue is more general, please create your issue here: [withfig/fig](https://github.com/withfig/fig/issues/new?assignees=&labels=NEED_TO_LABEL&template=main_issue_template.yml)
- [X] I have searched [github.com/withfig/autocomplete/issues](https://github.com/withfig/autocomplete/issues?q=) and there are no duplicates of my issue
- [X] I would like to work on this.

### What CLI tool does this relate to?

aws

### Which statement makes the most sense?

This completion spec isn't complete

### Issue Details

All AWS commands can be suffixed with `--profile PROFILE`, in which `PROFILE` is one of the configured profile and can be fetched using `aws configure list-profiles`.

I suggest Fig autocompletes the profile list on **all** aws commands

https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-profiles.html

### PR DESCRIPTION
**Pull request by joeheyming**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
In this commit, we want to make it so aws --profile auto completes
with the list of profiles you have available.

**What is the current behavior? (You can also link to an open issue here)**
Currently the aws command doesn't know about --profiles

**What is the new behavior (if this is a feature change)?**
Generally these profiles live inside a credentials file
but instead of directly looking at the credentials file,
we ask aws to give us a list of the profiles using
`aws configure list-profiles`

**Additional info:**

closes #542

### CODE REVIEW

File: src/aws.ts [hunk 1]
```diff
@@ -1,3 +1,32 @@
+export const awsProfileGenerator: Fig.Generator = {
+  cache: {
+    strategy: "stale-while-revalidate",
+    cacheByDirectory: true,
+  },
+  script: "aws configure list-profiles",
+  postProcess: function (out, [awsClient]) {
```
Comment (fedeci) [inline src/aws.ts:7]: nit:
```suggestion
  postProcess: function (out) {
```

File: src/aws.ts [hunk 2]
```diff
@@ -1,3 +1,32 @@
...
+          return {
+            name: line,
+          };
+        });
+      }
+    } catch (e) {
+      console.error(e);
+    }
+
+    return [];
```
Comment (fedeci) [inline src/aws.ts:26]: `trycatch` does nothing in this case and `String.prototype.split` always returns an array of at least one value if the string exists:
```suggestion
    const lines = out.split("\n");
    return lines.map((line) => {
      return {
        name: line,
      };
    });
```

**Review by fedeci — CHANGES_REQUESTED** [review]
Thanks for the improvement to this spec @joeheyming!

**Comment by joeheyming** [comment]
Screen Shot 2022-10-23 at 11 20 25 AM

**Review by fedeci — APPROVED** [review]
Thanks!

**Pull request merged**